fabric/tasks.py

```diff
@@ -411,14 +417,23 @@ def execute(task, *args, **kwargs):
             ran_jobs = jobs.run()
             for name, d in ran_jobs.iteritems():
                 if d['exit_code'] != 0:
-                    if isinstance(d['results'], NetworkError) and \
-                            _is_network_error_ignored():
-                        error(d['results'].message, func=warn, exception=d['results'].wrapped)
+                    if (
+                        isinstance(d['results'], NetworkError) and
+                        _is_network_error_ignored()
+                    ):
+                        error(
+                            d['results'].message, func=failed_instance,
+                            exception=d['results'].wrapped
+                        )
                     elif isinstance(d['results'], BaseException):
-                        error(err, exception=d['results'])
+                        failed = True
+                        error(name, func=failed_instance, exception=d['results'])
                     else:
-                        error(err)
+                        failed = True
+                        error(name + "experienced an unknown error", func=failed_instance)
                 results[name] = d['results']
+            if failed:
```

Comment:
This will suppress the warnings that should be given by the `NetworkError` check above.

I'd consider something like this, ±pep8:

```python
            if failedMessages:
                error_func = None if failed else warn
                error(err + "\n\n" + "\n".join(failedMessages) + "\n\n............................", func=error_func)
```